### PR TITLE
DDP-6657 pancan: add-child flow

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

This PR sets up the "Add Child" flow for PanCan. This is based on the similar flow in other studies such as RareX.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

- [x] :relaxed: All good, business as usual!

## How do we demo these changes?

Need to wait for frontend support, but in the meantime you can call the APIs directly.

## Testing

- [x] I have written zero automated tests but have poked around locally to verify proper functionality

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java [hunk 1]
```diff
@@ -36,9 +36,9 @@ int insert(@Bind("stateId") long stateId, @Bind("studyGuid") String studyGuid,
             + " where study_activity_id = :activityId and check_each_instance = :check")
     Optional<Long> findActivityStateId(@Bind("activityId") long activityId, @Bind("check") boolean checkEachInstance);
 
-    @SqlQuery("select workflow_state_id, study_guid, study_name, redirect_url from workflow_study_redirect_state "
-            + " where (study_guid is null or study_guid = :studyGuid) and (study_name is null or study_name = :studyName) "
-            + " and redirect_url = :redirectUrl")
+    @SqlQuery("select * from workflow_study_redirect_state"
+            + " where (study_guid = :studyGuid and redirect_url = :redirectUrl)"
+            + "    or (study_name = :studyName and redirect_url = :redirectUrl)")
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java:41]: The old query was not working for me (the add-child workflow adds same study redirect rules and it's creating duplicate rows). No idea why but this new query seems to work.

File: study-builder/studies/pancan/study-workflows.conf [hunk 2]
```diff
@@ -69,7 +69,7 @@
         {
           "type": "STUDY_REDIRECT",
           "studyGuid": null,
-          "studyName": "Leiosarcoma Project"
+          "studyName": "Leiomyosarcoma Project"
           "redirectUrl": ${lmsRedirectUrl},
           "expression": ${_pex.is_self}"&&"${_pex.has_lms},
```
Comment (yufengwng) [inline study-builder/studies/pancan/study-workflows.conf]: @ssettipalli Is this a bug? Do we need to redirect to LMS in the child-only case as well?

File: study-builder/studies/pancan/subs.conf [hunk 3]
```diff
@@ -259,6 +263,115 @@
         )
       ))
     """,
```
Comment (yufengwng) [inline study-builder/studies/pancan/subs.conf:265]: Sorry about the duplication -- it was the fastest way for me to get this done.

File: study-builder/studies/pancan/study-workflows.conf (hunk 2 above)
Comment (ssettipalli) [reply study-builder/studies/pancan/study-workflows.conf]: I believe we need to redirect for Leiomyosarcoma child too. ${_pex.is_self} can be removed.

File: study-builder/studies/pancan/study.conf [hunk 4]
```diff
@@ -81,56 +81,70 @@
...
     """,
+    # These rules apply to both normal registration and add-child flows, so check both.
     "ageOfMajorityRules": [
       # AoM for U.S.
       {
-        "condition": ${_pex.is_child_only}"&&"${_pex.is_country_us}"""
+        "condition": "("${_pex.has_prequal}"&&"${_pex.is_child_only}"&&"${_pex.is_country_us}"""
           && user.studies["cmi-pancan"].forms["PREQUAL"].questions["STATE"].answers.hasOption("AL")
+          ) || ("""${_pex.has_addchild}"&&"${_pex.addchild_is_country_us}"""
+          && user.studies["cmi-pancan"].forms["ADD_CHILD"].questions["CHILD_STATE"].answers.hasOption("AL"))
```
Comment (ssettipalli) [inline study-builder/studies/pancan/study.conf:96]: nice that we can include ADD_CHILD to the AOM rules

Comment (ssettipalli) [reply study-builder/studies/pancan/study.conf:96]: Assuming AOM triggers/events/actions like REACHED_AOM, GOVERNED_USER_REGISTERED , REVOKE_PROXIES should all fall in place for ADD_CHILD(s) or do we need to do any additional stuff for ADD_CHILD(s) !!

**Comment by ssettipalli** [comment]
Looks good to me

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java (hunk 1 above)
Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java:41]: @yufengwng was it create duplicates in unit tests or in dev env? May be somehow affects DB data that duplicated rows were created?

File: study-builder/studies/pancan/study.conf (hunk 4 above)
Comment (yufengwng) [reply study-builder/studies/pancan/study.conf:96]: The age-up events (`reached_aom`, `governed_user_registered`, etc) doesn't depend on `add_child` activity, so I don't see any issues there.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java (hunk 1 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiWorkflowState.java:41]: I think it somehow couldn't find the existing row in DB and tried to create another one and causing duplicate error, not sure why it couldn't find it though.

**Review by evgeniipr — DISMISSED** [review]

File: study-builder/studies/pancan/study-workflows.conf (hunk 2 above)
Comment (yufengwng) [reply study-builder/studies/pancan/study-workflows.conf]: Fixed!

**Review by ssettipalli — APPROVED** [review]

**Pull request merged**